### PR TITLE
Remove warning when redirecting to internal link

### PR DESCRIPTION
**Pull request by Rohan-cp**
### Summary of Changes

Provide a brief description of the changes made in this PR. Explain the problem you are fixing or the feature you are adding.

### Demo or Before/After Pics

- If this is a new feature, include screenshots or recordings of the feature in action.
- If this PR makes visual changes, include before and after screenshots.

| Before                                     | After                                      |
| ------------------------------------------ | ------------------------------------------ |
|  |  |

### Edge Cases

List any common edge cases that you have considered and tested.

### Testing Steps

Provide steps on how the reviewer can test the changes.

### Checklist

Please make sure to review and check all of the following:

- [ ] I've tested the changes and all tests pass.
- [ ] (if web) I've tested the changes on various desktop screen sizes to ensure responsiveness.
- [ ] (if mobile) I've tested the changes on both light and dark modes.

### CODE REVIEW

File: apps/mobile/src/utils/handleDeepLinkPress.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
+import { RootStackNavigatorProp } from '~/navigation/types';
+
+const KNOWN_NON_DEEPLINK_ROUTES = ['community', '~'];
+
+export async function handleDeepLinkPress(url: string, navigation: RootStackNavigatorProp) {
```
Comment (Robinnnnn) [inline apps/mobile/src/utils/handleDeepLinkPress.tsx:5]: hmm, it seems like we're duplicating a good deal of business logic across this function and the one in `DeepLinkRegistrar.tsx`. could we consolidate the code into a single helper?

File: apps/mobile/src/components/Markdown.tsx [hunk 2]
```diff
@@ -46,6 +49,8 @@ type GalleryMarkdownProps = PropsWithChildren<{
   style?: StyleProp<unknown>;
 }>;
 
+const KNOWN_NON_DEEPLINK_ROUTES = ['community', '~', 'home', 'token'];
```
Comment (Robinnnnn) [inline apps/mobile/src/components/Markdown.tsx:52]: imo this kind of deeplink-related logic should live in `DeepLinkRegistrar.tsx`, and `Markdown.tsx` should remain as "dumb" as possible

File: apps/mobile/src/components/Markdown.tsx [hunk 3]
```diff
@@ -103,14 +110,27 @@ export function Markdown({
...
-      if (url && onBypassLinkPress) {
+      const isInternalLink = url.startsWith('https://gallery.so/');
+      const parsedUrl = new URL(url);
+      const splitBySlash = parsedUrl.pathname.split('/').filter(Boolean);
+
+      const isInternalLinkWithDeepLink =
+        isInternalLink &&
+        !(
+          typeof splitBySlash[0] === 'string' && KNOWN_NON_DEEPLINK_ROUTES.includes(splitBySlash[0])
+        );
```
Comment (Robinnnnn) [inline apps/mobile/src/components/Markdown.tsx:121]: i think this check for `isInternalLinkWithDeepLink` could also be an external function exported out of `DeepLinkRegistrar`

File: apps/mobile/src/components/DeepLinkRegistrar.tsx [hunk 4]
```diff
@@ -46,13 +151,9 @@ export function DeepLinkRegistrar() {
         }
 
         const parsedUrl = new URL(url);
```
Comment (Robinnnnn) [inline apps/mobile/src/components/DeepLinkRegistrar.tsx:153]: toss this, marfa

**Review by Robinnnnn — APPROVED** [review]
